### ISSUE TITLE
Download mirror support

### ISSUE THREAD

**Issue by xepaul** [purescript/npm-installer#4]
Please can a custom binaries url be supported for use in the enterprise.

Setting a Path variable like  chromedriver would work well. 

Thanks

https://github.com/giggio/node-chromedriver/ 

https://www.npmjs.com/package/chromedriver

**Comment by hdgarrood** [comment]
Can you provide a bit more context on why this would be useful please?

**Comment by xepaul** [comment]
The current script can not be used when run behind a firewall, github.com is not available.
Thanks

**Comment by xepaul** [comment]
A local enterprise version of NPM is being used.

**Comment by xepaul** [comment]
A path needs to be able to be set to download pure script from a mirror instead of github.com like purescript_downloadUrl=https://npm.xyz.org/mirrors/purescript

**Comment by hdgarrood** [comment]
Right, that makes sense, thanks. Where are you imagining mirrors will be hosted though?

**Comment by xepaul** [comment]
Internally in the enterprise inside the firewall.

**Comment by hdgarrood** [comment]
So your plan is to get around the firewall each time there's a compiler release in order to download each of the binaries from the normal location, and then make them available internally? Fair enough.

When implementing this, if a mirror is set, we should ensure that we're using the source tarball from the same location too (i.e. not github) in the case where we fall back to building from source.

**Comment by xepaul** [comment]
Yes that’s correct the binaries would be made available internally. Yes seems logical to have the source on the fallback scenario.
